game_engine/ecs/systems/ServerInputSystem.cpp

```
@@ -0,0 +1,87 @@
+
+void ecs::ServerInputSystem::queueInput(Entity entityId,
+                                        const PlayerInput &input) {
+  for (const auto &pendingInput : _pendingInputs[entityId]) {
+    if (input.input == pendingInput.input) {
+      return;
+    }
+  }
+  _pendingInputs[entityId].push_back(input);
+}
+
+void ecs::ServerInputSystem::processInput(
```

Clang (14.0.6)

[warning] method 'processInput' can be made static

(readability-convert-member-functions-to-static)